### PR TITLE
Webhooks

### PR DESCRIPTION
**Pull request by mwasilew**
Add webhook support for jobserv

After the build is completed, if the project definition contains a webhook entry, URL defined in the webhook is called. To ensure it's jobserv that is calling, custom header X-JSrv-Signature-256 is added to the POST headers. POST payload contains the following fields:
 - project_name
 - build_id
 - build_status
 - build_url
 - build_reason
 - runs
Each run contains the following fields:
 - url
 - name
 - status

### CODE REVIEW

File: jobserv/notify.py [hunk 1]
```diff
@@ -132,6 +134,35 @@ def notify_build_complete(build, to_list):
...
+        payload['runs'].append({
+            'url':run_url(run),
+            'name': run.name,
+            'status': run.status.name
+            })
+    try:
+        requests.post(webhook_url, data=payload, headers=headers)
+    except requests.exceptions.RequestException:
+        # what to do here?
+        pass
```
Comment (doanac) [inline jobserv/notify.py:162]: I'm tempted to say you could fail the build. However, I'm about to recommend doing something I rarely recommend: This webhook call might be slow and its currently being from the context of an API request. We currently don't have any redis/rabbit type queue in this project. How about we create a thread and do the upload from that. You could maybe put some simple retry/sleep type logic and then if it fails, just log an error.

File: jobserv/notify.py [hunk 2]
```diff
@@ -132,6 +134,35 @@ def notify_build_complete(build, to_list):
     _send(msg)
 
 
+def notify_build_complete_webhook(build, webhook_url, secret):
+    # try to validate URL certificate before sending POST?
+    # add secret header
+    headers = {
+        'X-JSrv-Signature-256 ': hashlib.sha256(secret).hexdigest()
```
Comment (doanac) [inline jobserv/notify.py:141]: I think github has a slightly better approach for this:

https://github.com/foundriesio/jobserv/blob/master/jobserv/api/github.py#L137-L147

File: jobserv/api/run.py [hunk 3]
```diff
@@ -90,7 +90,14 @@ def _handle_build_complete(projdef, storage, build, params, secrets, trigger):
     if email:
         if build.status == BuildStatus.FAILED \
                 or not email.get('only_failures'):
-            notify_build_complete(build, email['users'])
+            notify_build_complete_email(build, email['users'])
+    if projdef.project_webhooks:
+        for webhook in webhooks:
+            if build.status == BuildStatus.FAILED \
+                    or not webhook.get('only_failures'):
+                secret = trigger.secret_data.get(webhook['secret_name'], None)
```
Comment (doanac) [inline jobserv/api/run.py:98]: think this needs to be `secret = secrets.get(webhook['secret_name'])`

**Comment by mwasilew** [comment]
Abandoning in favor of #37

**Pull request closed without merging**